### PR TITLE
Refactor compressors to be in separate files

### PR DESCRIPTION
**Pull request by asinghvi17**
This PR reformats the compressors to be in a folder with separate files, and documents the "compressor API".

(Previous commits added the fletcher32 filter as a compressor, but on review that should be a filter so I'll put it in a new PR.)

### CODE REVIEW

**Comment by asinghvi17** [comment]
I've implemented fletcher32 compression in this last commit, and am testing it with unit tests copied from the Python package.  I'm not sure what the best way to load it in integration testing is though, so could use a hand with that!

File: src/Compressors/Compressors.jl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+Finally, an entry MUST be added to the `compressortypes` dictionary for each compressor type.  
+This must also follow the Zarr specification's name for that compressor.  The name of the compressor
+is the key, and the value is the compressor type (e.g. `BloscCompressor` or `NoCompressor`).
+
+For example, the Blosc compressor is named "blosc" in the Zarr spec, so the entry for [`BloscCompressor`](@ref) 
+must be added to `compressortypes` as `compressortypes["blosc"] = BloscCompressor`.
+"""
+abstract type Compressor end
+
+compressortypes = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```
Comment (asinghvi17) [inline src/Compressors/Compressors.jl:39]: Should this be a `const`?  It's still mutable if it is but I feel like that makes it a bit clearer.

```suggestion
const compressortypes = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```

or, for maximum clarity,

```suggestion
const COMPRESORTYPES = Dict{Union{String,Nothing}, Type{<: Compressor}}()
```

**Comment by asinghvi17** [comment]
That's strange - tests are all passing, but the build failed when it came time to upload the codecov data, saying the job was already closed.

**Comment by asinghvi17** [comment]
> `Error: Bad response: 422 {"message":"Can't add a job to a build that is already closed. Build 10492471554 is closed. See docs.coveralls.io/parallel-builds","error":true}`

(this line repeats in every log from the failed CI)

**Review by meggart — APPROVED** [review]

**Comment by meggart** [comment]
Is this ready to be merged? Or do you want to fix something in the CI pipeline first?

**Comment by asinghvi17** [comment]
Let's merge this, I'll fix the pipeline in another PR.

**Pull request merged**